controllers/storagecluster/storagecluster_controller.go

```diff
@@ -107,6 +108,17 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
+	storageConsumerStatusPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectOld == nil || e.ObjectNew == nil {
+				return false
+			}
+			oldObj := e.ObjectOld.(*ocsv1alpha1.StorageConsumer)
+			newObj := e.ObjectNew.(*ocsv1alpha1.StorageConsumer)
+			return reflect.DeepEqual(oldObj.Status.Client, newObj.Status.Client)
```

Comment:
I believe this needs a negation on the return value

```suggestion
			return !reflect.DeepEqual(oldObj.Status.Client, newObj.Status.Client)
```